src/forecaster/heuristic.py

```diff
@@ -33,27 +52,52 @@ def apply_heuristics(internal_series: dict, enrolment: dict, low_bound: int, hig
     :return: None, the internal series is modified in place
     """
 
-    remaining_seats = high_bound
-    unassigned_courses = 0
-
     # Assign capacities to courses which have a data point
     for course in internal_series.keys():
         if internal_series[course]["capacity"] <= 0:
             for i, enrolment in enumerate(reversed(internal_series[course]["data"])):
                 if enrolment != 0:
-                    internal_series[course]["capacity"] = math.floor(enrolment * math.pow(PROGRAM_GROWTH, (i+1)))
+                    internal_series[course]["capacity"] = floor(enrolment * pow(PROGRAM_GROWTH, (i+1)))
                     break
 
-    # Compute number of remaining seats, and unassigned courses
+    # Assign capacities to courses which have no data point
+    # but are similar to a course with data
+    course_types = set()
+    for course in internal_series.keys():
+        if internal_series[course]["capacity"] <= 0:
+            course_types.add(get_course_type(course))
+
+    # Initialize average assignments to 0
+    average_assignments = {course_type: [0] for course_type in course_types}
+
+    # Calculate average assignments
+    for course in internal_series.keys():
+        type = get_course_type(course)
+        if type in course_types:
+            average_assignments[type].append(internal_series[course]["capacity"])
+    
+    for assignment in average_assignments.keys():
+        average = floor(average_capacity(average_assignments[assignment]))
+        print("Average of {} is {}".format(assignment, average))
+        average_assignments[assignment] = average
+
+    for course in internal_series.keys():
+        if internal_series[course]["capacity"] <= 0:
+            type = get_course_type(course)
+            if type in average_assignments.keys():
+                internal_series[course]["capacity"] = average_assignments[type]
+            
+    # Assign remaining courses via heuristic 3
     for course in internal_series.keys():
-        if internal_series[course]["capacity"] != 0:
-            remaining_seats -= internal_series[course]["capacity"]
-        else:
-            unassigned_courses += 1
-
-    # Assign remaining courses via Heuristic 2
-    if unassigned_courses > 0:
-        seats_per_course = math.floor(remaining_seats / unassigned_courses)
-        for course in internal_series.keys():
-            if internal_series[course]["capacity"] <= 0:
-                internal_series[course]["capacity"] = seats_per_course
+        if int(internal_series[course]["capacity"]) <= 0:
+            course_code = ''.join(c for c in course if c.isdigit())
+            if course_code[0] >= '4': # fourth year or greater
+                internal_series[course]["capacity"] = 50
```

Comment:
maybe add these constants to constants.py so they can easily be changed and configured

Reply:
That's a good point, I've made that change now.